.github/workflows/ci.yml

```diff
@@ -0,0 +1,101 @@
+name: CI
+
+on:
+  push:
+    # branches:
+    # - main
+  pull_request:
+    branches:
+    - main
+  # Run weekly at 1:02 Monday UTC
+  schedule:
+  - cron:  '2 1 * * 1'
+  workflow_dispatch:
+
+jobs:
+  test:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.9]
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install build dependencies on Ubuntu
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y \
+          ca-certificates \
+          lsb-release \
+          wget \
+          build-essential \
+          pkg-config \
+          cmake
+        wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+        sudo apt-get install -y ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+        sudo apt-get update -y
+        sudo apt-get install -y \
+          libarrow-dev \
+          libparquet-dev
+
+        # Install other tools
+        sudo apt-get install -y tree
+
+    - name: Install build dependencies on macOS
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew install apache-arrow
+        brew install tree
+
+    - name: Build library and examples on Ubuntu
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        cmake \
+          -DCMAKE_MODULE_PATH=$(find /usr/lib -type d -name arrow) \
+          -S . \
+          -B build
+        cmake build -L
+        cmake --build build --parallel $(($(nproc) - 1))
+        tree build
+
+    - name: Build library and examples on macOS
+      if: matrix.os == 'macos-latest'
+      run: |
+        ls -l /usr/local/Cellar/apache-arrow/*
+        cmake \
+          -DARROW_PATH=/usr/local/Cellar/apache-arrow \
+          -S . \
+          -B build
```

Comment:
At the moment the CI is failing because of the `cmake` command on maOS &mdash; if macOS is turned off the Linux workflow passes in full.

At the moment, the build logs give the following

```
-rw-r--r--   1 runner  wheel  1221676 Jul 22 12:40 CHANGELOG.md
-rw-r--r--   1 runner  wheel     4783 Sep 17 05:02 INSTALL_RECEIPT.json
-rw-r--r--   1 runner  wheel   108994 Jul 22 12:40 LICENSE.txt
-rw-r--r--   1 runner  wheel     3032 Jul 22 12:40 NOTICE.txt
-rw-r--r--   1 runner  wheel     5302 Jul 22 12:40 README.md
drwxr-xr-x   3 runner  wheel       96 Jul 22 12:40 bin
drwxr-xr-x   6 runner  wheel      192 Jul 22 12:40 include
drwxr-xr-x  37 runner  wheel     1184 Jul 22 12:40 lib
drwxr-xr-x   3 runner  wheel       96 Jul 22 12:40 share
-- The C compiler identification is AppleClang 12.0.0.12000032
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
CMake Error at cmake/FindArrow.cmake:111 (message):
  Could not find the Arrow library.  Looked for headers in
  /usr/local/Cellar/apache-arrow/include, and for libs in
  /usr/local/Cellar/apache-arrow/lib
Call Stack (most recent call first):
  CMakeLists.txt:17 (find_package)


-- Configuring incomplete, errors occurred!
```

I'm not sure how to properly find this library, as I have no real experience debugging macOS package management problems. The `brew install` step earlier is showing things getting installed though

```console
==> Installing apache-arrow
==> Pouring apache-arrow--5.0.0_1.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/apache-arrow/5.0.0_1: 545 files, 113.2MB
```

Reply:
TIL that `brew --prefix` exists and that macOS doesn't have `nproc` and you have to use `sysctl -n hw.logicalcpu` instead. :/